### DISCUSSION TITLE
Tasmota bulb goes offline and doesn’t reconnect

### DISCUSSION THREAD

**Discussion by ferbulous** [arendst/Tasmota#19534] [Support request]
Hi,I have a tasmota bulb running in a detached mode.

Lately it’s been disconnecting more frequently and doesn’t reconnect even with this rule added that reboots tasmota when wifi is disconnected.
I had to power cycle to get it online again when that happens

Rule 
```
Rule1
ON wifi#connected DO ruletimer1 0 ENDON  ON wifi#disconnected DO ruletimer1 60 ENDON  ON rules#timer=1 DO restart 1 ENDON
``` 

Device details
``` 

Program Version	13.1.0(tasmota32)
Build Date & Time	2023-08-15T08:01:16
Core/SDK Version	2_0_11/4.4.5.230722
Uptime	0T00:04:08
Flash Write Count	1810
Boot Count	391
Restart Reason	Vbat power on reset
Friendly Name 1	Tasmota
 
AP1 SSId (RSSI)	homewifi (74%, -63 dBm) 11n
Hostname	tasmota-635FD4-8148
IPv6 Global (wifi)	2001:d08:dc:8197:7edf:a1ff:fe63:5fd4
IPv6 Local (wifi)	fe80::7edf:a1ff:fe63:5fd4
MAC Address	7C:DF:A1:63:5F:D4
IP Address (wifi)	192.168.1.3
Gateway	192.168.1.254
Subnet Mask	255.255.255.0
DNS Server1	8.8.8.8
DNS Server2	8.8.4.4
 
HTTP API	Enabled
 
MQTT Host	192.168.1.40
MQTT Port	1883
MQTT TLS	Disabled
MQTT User	tasmota2
MQTT Client	DVES_635FD4
MQTT Topic	tasmota_%06X
MQTT Group Topic 1	cmnd/tasmotas/
MQTT Full Topic	cmnd/tasmota_635FD4/
MQTT Fallback Topic	cmnd/DVES_635FD4_fb/
MQTT No Retain	Disabled
 
Emulation	None
mDNS Discovery	Disabled
 
ESP Chip Id	6512596 (ESP32-C3 rev.3)
Flash Chip Id	0x164020 (DIO)
Flash Size	4096 KB
Program Flash Size	4096 KB
Program Size	1854 KB
Free Program Space	897 KB
Free Memory	120.8 KB (frag. 33%)
 
Partition safeboot	832 KB (used 91%)
Partition app0*	2752 KB (used 67%)
Partition fs	444 KB
``` 

Ha log
![E901EC4C-3851-4D22-8BBB-7ACBE32082B9](https://github.com/arendst/Tasmota/assets/24526737/3d5259f4-72ad-416e-8bcd-a44e1e8b6e7e)

Any thoughts on this?

**Comment by Jason2866** [comment]
Without any device log data when this happens not possible to tell what really happens.
You can try latest development version (Arduino Core updates).
Good wifi rssi? Take a look in Tasmota info page. Bad wifi ends always in disconnects.

**Reply by ferbulous** [reply]
Is this acceptable?

AP1 SSId (RSSI)	homewifi (74%, -63 dBm) 11n

Looks like the device just went offline a minute ago, i’ll try updating to the development build and see if it improves.
How would i retrieve the logs?  Do i just connect tx/rx to raspberry pi with the bulb wired to the main?

**Reply by Jason2866** [reply]
NEVER connect anything to your computer or to other devices when the esp device is connected with AC mains!
You have to setup a syslog server. Tasmota does support.

74% wifi is okay.

**Reply by ferbulous** [reply]
Instead of connecting to AC mains, i need to open it up, connect an external 3.3v power supply as well tx/rx from the uart with syslog server set up on the raspbery pi?

I think I understand now, if I have the bulb wired to the ac main, Set the logging to level 4 with syslog server configured. Then I should get some error logs right before it disconnects from wifi?
